tools/slang-test/slang-test-main.cpp

```diff
@@ -4681,6 +4681,7 @@ static SlangResult runUnitTestModule(
                 {
                     String output = getOutput(exeRes);
                     reporter->message(TestMessageType::TestFailure, output.getBuffer());
+                    context->failedUnitTests.add(test.testName);
```

Comment:
if this is first attempt, we don't want to add a failure to reporter, since this will cause the entire test process to return failure even if retry succeeds.